bezier-rs/lib/src/lib.rs

```diff
@@ -122,57 +135,86 @@ impl Bezier {
 
 	pub fn get_handle1(&self) -> DVec2 {
 		match self.handles {
-			BezierHandles::Quadratic { handle } => {
-				handle
-			}
-			BezierHandles::Cubic { handle1, handle2: _ } => {
-				handle1
-			}
+			BezierHandles::Quadratic { handle } => handle,
+			BezierHandles::Cubic { handle1, handle2: _ } => handle1,
 		}
 	}
 
 	pub fn get_handle2(&self) -> Option<DVec2> {
 		match self.handles {
-			BezierHandles::Quadratic { handle: _ } => {
-				None
-			}
-			BezierHandles::Cubic { handle1: _, handle2 } => {
-				Some(handle2)
-			}
+			BezierHandles::Quadratic { handle: _ } => None,
+			BezierHandles::Cubic { handle1: _, handle2 } => Some(handle2),
 		}
 	}
 
-
 	pub fn get_points(&self) -> [Option<DVec2>; 4] {
 		match self.handles {
 			BezierHandles::Quadratic { handle } => [Some(self.start), Some(handle), Some(self.end), None],
 			BezierHandles::Cubic { handle1, handle2 } => [Some(self.start), Some(handle1), Some(handle2), Some(self.end)],
 		}
 	}
 
-	/// Return the length of the bezier curve
-	pub fn length() -> i32 {
-		0
-	}
-}
+	///  Calculate the x-value of a point on the curve based on the t-value provided
+	///  basis code based off of pseudocode found here: https://pomax.github.io/bezierinfo/#explanation
+	pub fn get_x_basis(&self, t: f64) -> f64 {
+		let t2 = t * t;
+		let mt = 1.0 - t;
+		let mt2 = mt * mt;
 
-/*
+		match self.handles {
+			BezierHandles::Quadratic { handle } => mt2 * self.start[0] + 2.0 * mt * t * handle[0] + t2 * self.end[0],
+			BezierHandles::Cubic { handle1, handle2 } => {
+				let t3 = t2 * t;
+				let mt3 = mt2 * mt;
+				mt3 * self.start[0] + 3.0 * mt2 * t * handle1[0] + 3.0 * mt * t2 * handle2[0] + t3 * self.end[0]
+			}
+		}
+	}
 
-/// for computing the length of a bezier curve
-/// taken from https://pomax.github.io/bezierinfo/#arclength
+	///  Calculate the y-value of a point on the curve based on the t-value provided
+	///  basis code based off of pseudocode found here: https://pomax.github.io/bezierinfo/#explanation
+	pub fn get_y_basis(&self, t: f64) -> f64 {
+		let t2 = t * t;
+		let mt = 1.0 - t;
+		let mt2 = mt * mt;
 
-computeLength(curve) {
-	const z = 0.5, len = T.length;
-	let sum = 0;
-	for (let i = 0, t; i < len; i++) {
-	  t = z * T[i] + z;
-	  sum += C[i] * this.arcfn(t, curve.derivative(t));
+		match self.handles {
+			BezierHandles::Quadratic { handle } => mt2 * self.start[1] + 2.0 * mt * t * handle[1] + t2 * self.end[1],
+			BezierHandles::Cubic { handle1, handle2 } => {
+				let t3 = t2 * t;
+				let mt3 = mt2 * mt;
+				mt3 * self.start[1] + 3.0 * mt2 * t * handle1[1] + 3.0 * mt * t2 * handle2[1] + t3 * self.end[1]
```

Comment:
Also instead of `some_dvec2[0]` and `some_dvec2[1]` you can do `some_dvec2.x` and `some_dvec2.y` which is probably easier to read.